helpers.lua

```diff
@@ -1065,6 +1057,15 @@ end
 -- Set cookie and redirect (needed to properly set cookie)
 function redirect(url)
     logger.debug("Redirecting to "..url)
+    -- For security reason we don't allow to redirect onto unknown domain
+    -- And if `uri_args.r` contains line break, someone is probably trying to
+    -- pass some additional headers
```

Comment:
Not tested but I'm also proposing to make sure that the url is prefixed with `https?://`, otherwise the regex may fail to extract the domain

```suggestion
    -- pass some additional headers
    -- This should cover the following cases:
    -- https://malicious.domain.tld/foo/bar
    -- http://malicious.domain.tld/foo/bar
    -- https://malicious.domain.tld:1234/foo
    -- malicious.domain.tld/foo/bar
    if not string.starts(url, "http://") and not string.starts(url, "https://") then
        url = "https://"..url
    end
```

Reply:
but this doesn't work with uri...

Reply:
Indeed, I added a `not string.starts(url, "/")`